### PR TITLE
Google cloud sdk pythonenv

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

Previously only direct dependencies were available in the python search
path. Transitive dependencies would not be available and thus a few
features would not work due to import errors. Those import errors were
being caugth and gave a (wrong) hint why it wouldn't work. By using
`python.withPackages` instead of the crafted search path with only the
direct dependencies those missing feature should now be working.

See issue #67094 for details on the observed errors.

I only verified that the command now doesn't complain about the missing pyopenssl. Maybe @toraritte or @red-hood could give it a try?

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/admin/google-cloud-sdk/default.nix [hunk 1]
```diff
@@ -53,8 +53,8 @@ in stdenv.mkDerivation rec {
         programPath="$out/google-cloud-sdk/bin/$program"
         binaryPath="$out/bin/$program"
         wrapProgram "$programPath" \
-            --set CLOUDSDK_PYTHON "${python}/bin/python" \
-            --prefix PYTHONPATH : "${pythonPath}"
+            --set CLOUDSDK_PYTHON "${pythonEnv}/bin/python" \
+            --prefix PYTHONPATH : "${pythonEnv}/${python.sitePackages}"
```
Comment (FRidh) [inline pkgs/tools/admin/google-cloud-sdk/default.nix]: Does it really need `PYTHONPATH`? We already point the the interpreter, is that not enough?

File: pkgs/tools/admin/google-cloud-sdk/default.nix [hunk 2]
```diff
@@ -13,7 +13,7 @@
 let
   pythonInputs = [ cffi cryptography pyopenssl crcmod ]
                  ++ lib.optional (with-gce) google-compute-engine;
-  pythonPath = lib.makeSearchPath python.sitePackages pythonInputs;
+  pythonEnv = python.withPackages (_: pythonInputs);
```
Comment (FRidh) [inline pkgs/tools/admin/google-cloud-sdk/default.nix]: This is OK for the application case, but not when using as library.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/google-cloud-sdk/default.nix (hunk 1 above)
Comment (andir) [reply pkgs/tools/admin/google-cloud-sdk/default.nix]: Their wrapper does a bunch of mangling with the execution environment. I can try with out it. I need some feedback from the actual users of the tool to see if it still works afterwards.

**Comment by FRidh** [comment]
I've cherry-picked the openssl change to staging.

**Comment by red-hood** [comment]
Creating signed URLs with gsutil works for me now with the changes in this PR. Thank you! I did not observer any regressions so far.

File: pkgs/tools/admin/google-cloud-sdk/default.nix (hunk 2 above)
Comment (andir) [reply pkgs/tools/admin/google-cloud-sdk/default.nix]: But this isn't a python library. Am I missing something?

This is just a bundle of CLI utils as far as I can tell.

Comment (FRidh) [reply pkgs/tools/admin/google-cloud-sdk/default.nix]: yet it is in `python-modules.nix`. If it is not usable as a library it should not be here.

Comment (andir) [reply pkgs/tools/admin/google-cloud-sdk/default.nix]: Alright. I removed it from there and changed the expression to just take an abitrary `python` argument that will be used to construct the required environment.

**Comment by red-hood** [comment]
Any chance this might get merged soon? It would be really helpful, the package as it is currently released is unfortunately broken and of limited use.

**Pull request merged**